corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -483,8 +477,7 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
             } else {
                 var selectedValues = {};
             }
-            selectedValues[self.selectedValuesKey] = self.selectedCaseIds.join(',');
-            selectedValues['current'] = self.selectedCaseIds.join(',');
+            selectedValues[sessionStorage.queryKey] = self.selectedCaseIds.join(',');
```

Comment:
I still think this can be removed from here and moved to the URL object: https://github.com/dimagi/commcare-hq/blob/81a7268717686bc48a75c909367d4111a12b242a/corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/util.js#L183

Reply:
What I mean is that this class should not need to maintain the local storage at all. Apart from loading values from local storage on init this class should not touch `sessionStorage`. It already maintains it's own list of case IDs and when the user clicks 'continue' we pass the list of selected IDs with the event:
```
FormplayerFrontend.trigger("menu:select", this.selectedCaseIds);
```
In the event handler we call `urlObject.addSelection(index);` (where `index` is the list of case IDs). 

I'm suggesting that we move the code that populates the `sessionStorage.selectedValues` into the `addSelection` method of the `CloudcareUrl` object.

Reply:
@snopoke Thanks for clarifying! The issue I've run into is that 
FormplayerFrontend.trigger("menu:select", this.selectedCaseIds);
is only run when continue is pressed and not a pagination action so I found that the `this.selectedCaseIds` is not passes outside the class on pagination actions which is why the sessionStorage actions happened here to be added to options [here](https://github.com/dimagi/commcare-hq/pull/31997/files/73e193e661a0cb9e5d47d80967d024ec0f8ba96c..14a7e4768884826684ef61206b78f825af028cab#diff-e9a325527828d83ca273a6ebef7ca67882d9dee8484486bbfcea88215b8c8531R154). I'll continue to investigate

Reply:
OK thanks, that makes sense. In that case I think it's fine to leave it as is for now. I'll take a look at the pagination workflow in more detail another time.

Reply:
Added this change: https://github.com/dimagi/commcare-hq/pull/31997/commits/925b50d5e9c791936b71385f9d08a05ed8cdc0c6 to move to the pagination option so the `sessionStorage` is only modified on pagination options rather than every checkbox selection change.

Reply:
With this change I won't you also need to update sessionStorage on the `"menu:select"` event?

Reply:
yes, took your suggestion to add to `addSelection` for this case when the selections are modified but a pagination event has not occurred since those changes: [3b5c3c8](https://github.com/dimagi/commcare-hq/pull/31997/commits/3b5c3c8779f8123d38c15fcb2e00643084c43fcf)